### ISSUE TITLE
[Feat]: Update to latest RestSharp 110

### ISSUE THREAD

**Issue by MaxwellDAssistek** [EasyPost/easypost-csharp#435]
### Feature Request Is New

- [X] I have verified that the requested feature does not already exist or has not already been requested.

### Description of the feature

We use the latest version of RestSharp in our systems and we would prefer to avoid downgrading.

**Comment by nwithan8** [comment]
Hello, thanks for reaching out!

We are actually right now in the process of dropping RestSharp as a dependency from our library, since it's not being used. That should remove the dependency conflict from your system. We'll keep you updated on the process!

**Comment by nwithan8** [comment]
Hey @MaxwellDAssistek , wanted to let you know we are in officially in the process of dropping RestSharp in our next major release (along with a number of breaking changes and improvements).

While we encourage you to update the the latest version when it gets released, in the meantime, we've released [v4.6.1](https://github.com/EasyPost/easypost-csharp/releases/tag/v4.6.1), which simply extends the RestSharp dependency range so you'll be able to use RestSharp v110 in your solution.

I'm going to go ahead and close this ticket, but let us know if there's anything else you need help with!

**Comment by MaxwellDAssistek** [comment]
@nwithan8 Unfortunately, I get the following exception when using the latest RestSharp:

```
Error: System.MissingMethodException: Method not found: 'Void RestSharp.RestClient..ctor(System.Net.Http.HttpClient, RestSharp.RestClientOptions, Boolean)'.
  at EasyPost._base.EasyPostClient..ctor(String apiKey, String baseUrl, HttpClient customHttpClient)
  at EasyPost.Client..ctor(String apiKey, String baseUrl, HttpClient customHttpClient)
  at LogisticsBusinessLogic.Services.ShippingService.WebhookTest() in <SNIP>\ShippingService.cs:line xx
```

Downgrading RestSharp to 108.0.1 fixes this.

**Comment by nwithan8** [comment]
We'll look into this and see if we can fix it. Thanks for reporting.

**Comment by MaxwellDAssistek** [comment]
@nwithan8 We are using .NET 7.

**Comment by dobbamca** [comment]
hi
any update on this?

**Comment by nwithan8** [comment]
@MaxwellDAssistek Hey Max, it seems there was some breaking change between v108 and v109 of RestSharp that makes it incompatible with our library. We've recalled `v4.6.1` and replaced it with [`v4.6.2`](https://github.com/EasyPost/easypost-csharp/releases/tag/v4.6.2), which still locks to RestSharp v108.

In your case, if downgrading to RestSharp v108 is not an option, we just released [`v5.0.0`](https://github.com/EasyPost/easypost-csharp/releases/tag/v5.0.0) of our library, which drops RestSharp as a dependency entirely. It is a major version bump, however, and has some significant breaking changes. You can see our [Upgrade Guide](https://github.com/EasyPost/easypost-csharp/blob/v5.0.0/UPGRADE_GUIDE.md) to see what updating to the new version would entail.
